operator/internal/central/extensions/reconcile_defaulting.go

```diff
@@ -87,24 +87,33 @@ func setDefaultsAndPersist(ctx context.Context, logger logr.Logger, central *pla
 	// This updates central both on the cluster and in memory, which is crucial since this object is used for the final
 	// updating within helm-operator and we have concurrently running controllers (the status controller),
 	// whose changes we must preserve.
-	err := client.Patch(ctx, central, patch)
+	logger.Info("updating defaulting annotations on Central object")
+	err := client.Patch(ctx, u, patch)
 	if err != nil {
 		return errors.Wrap(err, "patching Central annotations")
 	}
-
 	logger.Info("patched Central object",
-		"oldResourceVersion", baseCentral.GetResourceVersion(),
+		"oldResourceVersion", uBase.GetResourceVersion(),
 		"newResourceVersion", central.GetResourceVersion(),
```

Comment:
Shouldn't this be
```suggestion
		"newResourceVersion", u.GetResourceVersion(),
```
?